src/OpenTelemetry/Metrics/Exemplar/SimpleExemplarReservoir.cs

```diff
@@ -0,0 +1,149 @@
+// <copyright file="SimpleExemplarReservoir.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+
+namespace OpenTelemetry.Metrics;
+
+/// <summary>
+/// The SimpleExemplarReservoir implementation.
+/// </summary>
+internal sealed class SimpleExemplarReservoir : ExemplarReservoir
+{
+    private readonly int poolSize;
+    private readonly Random random;
+    private readonly Exemplar[] runningExemplars;
+    private readonly Exemplar[] tempExemplars;
+
+    private long measurementsSeen;
+
+    public SimpleExemplarReservoir(int poolSize)
+    {
+        this.poolSize = poolSize;
+        this.runningExemplars = new Exemplar[poolSize];
+        this.tempExemplars = new Exemplar[poolSize];
+        this.measurementsSeen = 0;
+        this.random = new Random();
+    }
+
+    public override void Offer(long value, ReadOnlySpan<KeyValuePair<string, object>> tags, int index = default)
+    {
+        this.Offer(value, tags);
+    }
+
+    public override void Offer(double value, ReadOnlySpan<KeyValuePair<string, object>> tags, int index = default)
+    {
+        this.Offer(value, tags);
+    }
+
+    public override Exemplar[] Collect(ReadOnlyTagCollection actualTags, bool reset)
+    {
+        for (int i = 0; i < this.runningExemplars.Length; i++)
+        {
+            this.tempExemplars[i] = this.runningExemplars[i];
+            if (this.runningExemplars[i].FilteredTags != null)
+            {
+                // TODO: Better data structure to avoid this Linq.
+                // This is doing filtered = alltags - storedtags.
+                // TODO: At this stage, this logic is done inside Reservoir.
+                // Kinda hard for end users who write own reservoirs.
+                // Evaluate if this logic can be moved elsewhere.
+                // TODO: The cost is paid irrespective of whether the
+                // Exporter supports Exemplar or not. One idea is to
+                // defer this until first exporter attempts read.
+                this.tempExemplars[i].FilteredTags = this.runningExemplars[i].FilteredTags.Except(actualTags.KeyAndValues.ToList()).ToList();
+            }
+
+            if (reset)
+            {
+                this.runningExemplars[i].Timestamp = default;
+                this.measurementsSeen = 0;
```

Comment:
Shouldn't this be reset on every collection cycle regardless of the MetricReader temporality?

Reply:
TBH, i am not very sure if thats the case. Adding something to clarify this with the spec.